### PR TITLE
[android] Fix tracking mode + camera race condition

### LINKED ISSUE #8977: MapBox.animateCamera or moveCamera could be fail in tracking mode
**Platform:Android**
**Mapbox SDK version: V5.0.2**

### Steps to trigger behavior

 1. Set Mapbox in tracking 
 2. Then do MapBox.animateCamera or moveCamera 

### Expected behavior
The map should be changed to new set position and the Mapbox shouldn't be in tracking mode any more

### Actual behavior
Sometime the mapbox still rest in tracking mode, then the map is alway on the current GPS position.

noted that It works well on the Mapbox V4.x

### PR DESCRIPTION
**Pull request by Guardiola31337**
Fixes #8977 

cc/ @tobrun

### CODE REVIEW

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java [hunk 1]
```diff
@@ -34,6 +34,7 @@
   private CameraPosition cameraPosition;
   private MapboxMap.CancelableCallback cameraCancelableCallback;
   private MapboxMap.OnCameraChangeListener onCameraChangeListener;
+  private boolean isFromLocation = false;
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/Transform.java]: we don't need to keep explicit state around isFromLocation

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java [hunk 2]
```diff
@@ -306,7 +306,7 @@ public boolean isDismissTrackingModesForCameraPositionChange() {
    * <p>
    * </p>
    *
-   * @param willAllowToDismiss True will allow animated camera changes dismiss a trackig mode
+   * @param willAllowToDismiss True will allow animated camera changes dismiss a tracking mode
    */
   public void setDismissTrackingModeForCameraPositionChange(boolean willAllowToDismiss) {
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: both setter and getter for `isResetTrackingWithCameraPositionChange` can be removed completely

**Comment by Guardiola31337** [comment]
@tobrun ready for a second round 😉

File: platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java [hunk 3]
```diff
@@ -280,36 +276,13 @@ void resetTrackingModesIfRequired(boolean translate, boolean rotate) {
...
-   * <p>
-   * </p>
-   *
-   * @param willAllowToDismiss True will allow animated camera changes dismiss a trackig mode
-   */
-  public void setDismissTrackingModeForCameraPositionChange(boolean willAllowToDismiss) {
-    isResetTrackingWithCameraPositionChange = willAllowToDismiss;
+  void resetTrackingModesIfRequired(CameraPosition currentCameraPosition, CameraPosition targetCameraPosition,
+                                    boolean isFromLocation) {
+    resetTrackingModesIfRequired(currentCameraPosition.target != targetCameraPosition.target, false, isFromLocation);
```
Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: use equals when comparing objects?

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: @tobrun that would mean resetting tracking when zooming, which is an issue ([#9105](https://github.com/mapbox/mapbox-gl-native/issues/9105)).

Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: I'm talking about the LatLng, not the CameraPosition object ;)

Comment (tobrun) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: `!currentCameraPosition.target.equals(targetCameraPosition.target)`

Comment (Guardiola31337) [inline platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/TrackingSettings.java]: @tobrun 😂 you're right! 👍

**Comment by LukasPaczos** [comment]
It looks great! Zooming in and out agressively don't reset tracking mode anymore.

One thing I'd change though is the value of `MapboxMap#easeCamera()` parameter `isDismissable`. It should take `false` if we do not want tracking mode to be dismissed and `true` otherwise. It's opposite of that now and it's a bit confusing.

I'd also block dismissing by default in 
```
public final void easeCamera(final CameraUpdate update, final int durationMs, final boolean easingInterpolator,
                               final MapboxMap.CancelableCallback callback) {
    easeCamera(update, durationMs, easingInterpolator, callback, false);
  }
```
but that's just a preference I guess.

**Comment by Guardiola31337** [comment]
@LukasPaczos 

> Zooming in and out agressively don't reset tracking mode anymore.

Is this ☝️ still happening? If so, could you open a new ticket?
Thanks a lot!

**Comment by LukasPaczos** [comment]
@Guardiola31337 It's expected behaviour, so I think we're good here :)

**Pull request merged**